scalableminds/webknossos
# Export visible trees as nml in frontend

Missing:
- [x] spinner while exporting
- [x] correct name for the nml 

### Mailable description of changes:
 - [TracingView] It is now possible to download an nml file of the currently visible trees. Select `Export as nml` from the dropdown in the trees tab.

### Steps to test:
- Create a tracing with multiple trees, a user bounding box, comments and branchpoints. Select `Export as nml` from the dropdown in the trees tab and save the nml. Select `Download` from the dropdown next to the save button and save the nml. Compare the two nmls -> they should be equal.
- Hide some of the trees via their checkmark and export as nml again. The hidden trees should not be part of the nml.
- Open a very big tracing, e.g. 200.000 nodes, the export functionality should still work reasonably fast.
- The nml name should be the same for the download and export functionality.

### Issues:
- fixes #2094

------
